pkg/config/tenancy/flags.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2022 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tenancy
+
+import (
+	"flag"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+const (
+	tenancyEnabled = "multi_tenancy.enabled"
+	tenancyHeader  = "multi_tenancy.header"
+	validTenants   = "multi_tenancy.tenants"
+)
+
+// AddFlags adds flags for tenancy to the FlagSet.
+func AddFlags(flags *flag.FlagSet) {
+	flags.Bool(tenancyEnabled, false, "Enable tenancy header when receiving or querying")
+	flags.String(tenancyHeader, "x-tenant", "HTTP header carrying tenant")
```

Comment:
As a first version, we could have a single flag `tenant-header` which would enable tenancy if it is defined. 

Internally we will most likely not use the predefined list of tenants (e.g. in the observatorium project) and if the storage layer does not need to have access to the tenant names I find this configuration redundant.